File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/UserViewModel.ts

export interface IUserViewModel  {
    email: string;
    phoneNumber: string;
    firstName: string;
    lastName: string;
    middleName: string;
    address: string;
    address2: string;
    postalCode: string;
    city: string;
}
export class UserViewModel  implements IUserViewModel {
    email: string;
    phoneNumber: string;
    firstName: string;
    lastName: string;
    middleName: string;
    address: string;
    address2: string;
    postalCode: string;
    city: string;
}

File: ProjectStart/ProjectStart.ViewModel/AccountViewModels/UserViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace ProjectStart.ViewModel.AccountViewModels
{
    public class UserViewModel
    {
        [Required]
        [EmailAddress]
        [Display(Name = "Email")]
        public string Email { get; set; }

        [Required]
        [DataType(DataType.PhoneNumber)]
        [Display(Name = "Phone Number")]
        public string PhoneNumber { get; set; }

        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        public string MiddleName { get; set; }

        [Required]
        public string Address { get; set; }
        public string Address2 { get; set; }
        [Required]
        [Display(Name = "Postal Code")]
        public string PostalCode { get; set; }
        [Required]
        public string City { get; set; }
    }
}
